prime-dotnet-webapi/Services/ReportingService.cs

```diff
@@ -0,0 +1,119 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Prime.Models;
+using Prime.HttpClients;
+
+namespace Prime.Services
+{
+    public class ReportingService : BaseService, IReportingService
+    {
+        private const int CalculationPeriodInDays = 14;
+
+        private readonly ICollegeLicenceClient _collegeLicenceClient;
+
+        public ReportingService(
+            ApiDbContext context,
+            ILogger<ReportingService> logger,
+            ICollegeLicenceClient collegeLicenceClient)
+            : base(context, logger)
+        {
+            _collegeLicenceClient = collegeLicenceClient;
+        }
+
+        public async Task<string> PopulatePractitionerTableAsync(DateTime? startDate = null, DateTime? endDate = null)
+        {
+            var result = "";
+            try
+            {
+                if (startDate == null)
+                {
+                    startDate = DateTime.Now.AddDays(-CalculationPeriodInDays);
+                }
+
+                if (endDate == null)
+                {
+                    endDate = DateTime.Now;
+                }
+
+                // get all approved enrollee
+                var enrolleeLicences = _context.Enrollees
+                    .Where(e => e.GPID != null && e.Certifications.Any(c => c.Prefix != null))
+                    .Select(e => new
+                    {
+                        e.Certifications.FirstOrDefault().LicenseNumber,
```

Comment:
For nurses and midwives, I think `e.Certifications.FirstOrDefault().PractitionerId` is what we should compare against to `PharmanetTransactionLog.PractitionerId`